### PR TITLE
[android][ncl] QA fixes for Contacts module

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

Fixes one of bugs listed here #3010 

# How

- Fixed incorrect parameter types of `addContactAsync` and `removeContactAsync`
- Added missing promise reject calls
- Added asking for Android's `WRITE_CONTACTS` permission
- Applied some AS suggestions
- Fixed NCL Contacts screen not refreshing correctly after deleting a contact

# Test plan

Tested Contacts screen in NCL on Android.

### CODE REVIEW

**Comment by esamelson** [comment]
@EvanBacon - do we want the `WRITE_CONTACTS` permission on Android, or was that left out intentionally?

File: packages/expo-permissions/android/src/main/java/expo/modules/permissions/PermissionsModule.java [hunk 1]
```diff
@@ -119,6 +119,7 @@ public void askAsync(final ArrayList<String> requestedPermissionsTypes, final Pr
           break;
         case "contacts":
           permissionsTypesToBeAsked.add(Manifest.permission.READ_CONTACTS);
+          permissionsTypesToBeAsked.add(Manifest.permission.WRITE_CONTACTS);
```
Comment (EvanBacon) [inline packages/expo-permissions/android/src/main/java/expo/modules/permissions/PermissionsModule.java]: @jesseruder had a concern about this permission being too evasive.

**Review by EvanBacon — APPROVED** [review]

**Comment by EvanBacon** [comment]
Looks good. Just curious if @jesseruder has any thoughts on adding the permission now. I'll ping him

**Comment by esamelson** [comment]
I am concerned about adding this permission to the default list -- lots of people don't set `android.permissions` in their app.json and so they would get this new permission added by default, which generates a warning in the play store console.

Afaik there haven't been a ton of people asking for these features on Android, so could we compromise by keeping them available if you eject and add the WRITE_CONTACTS permission manually to your own AndroidManifest, but not including this permission in any template files or on any turtle builds?

**Review by sjchmiela — APPROVED** [review]

**Comment by tsapeta** [comment]
@esamelson @EvanBacon 
Considering your concerns about adding `WRITE_CONTACTS` as a default permission in the manifest, I've made some changes to allow users to add this permission through `android.permissions` config in `app.json` and when someone asks for `Permissions.CONTACTS` we will ask for both permissions in runtime but only if `android.permissions.WRITE_CONTACTS` is present in AndroidManifest.
See another related PR: https://github.com/expo/expo-cli/pull/265

@EvanBacon could you update docs for contacts including the fact that some methods require this permission? Also, some methods still have `iOS Only - temporary` there.

**Review by esamelson — APPROVED** [review]
Seems good, thanks @tsapeta !!

**Pull request merged**